### PR TITLE
GCW 2119 improvements for orders search (aka designations)

### PR DESCRIPTION
**Pull request by patrixr**
Changes  : 
- Addition of tests for existing api pagination
- Orders search is now extended to descriptions, organisations, user names
- Orders search is now case insensitive

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -241,9 +241,14 @@ def add_to_stockit
...
-      stockit_local_orders.client_name LIKE :query OR
-      stockit_contacts.first_name LIKE :query OR stockit_contacts.last_name LIKE :query OR
+    .where(" code LIKE :query OR 
+      description ILIKE :query OR
+      organisations.name_en ILIKE :query OR
+      organisations.name_zh_tw ILIKE :query OR
+      CONCAT(users.first_name, ' ', users.last_name) ILIKE :query OR
+      stockit_organisations.name ILIKE :query OR
+      stockit_local_orders.client_name ILIKE :query OR
+      stockit_contacts.first_name ILIKE :query OR stockit_contacts.last_name LIKE :query OR
```
[the remark this answers is not in the document]
Comment (patrixr) [reply app/models/order.rb]: Adding it now :)

File: app/models/order.rb [hunk 2]
```diff
@@ -241,9 +241,14 @@ def add_to_stockit
 
   def self.search(search_text, to_designate_item)
     fetch_orders(to_designate_item)
-    .where(" code LIKE :query OR stockit_organisations.name LIKE :query OR
-      stockit_local_orders.client_name LIKE :query OR
-      stockit_contacts.first_name LIKE :query OR stockit_contacts.last_name LIKE :query OR
+    .where(" code ILIKE :query OR
```
Comment (namrataukirde) [inline app/models/order.rb]: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.  …

code climate warning.

File: app/controllers/api/v1/orders_controller.rb [hunk 3]
```diff
@@ -40,7 +40,7 @@ def index
         return my_orders if is_browse_app?
         return recent_designations if params['recently_used'].present?
         records = @orders.with_eager_load.
-          search(params['searchText'], params['toDesignateItem'].presence).latest.
+          search(params['searchText'], params['toDesignateItem'].presence).descending.
```
Comment (namrataukirde) [inline app/controllers/api/v1/orders_controller.rb]: Place the . on the next line, together with the method name.  

code climate warning.

File: app/models/order.rb [hunk 4]
```diff
@@ -40,7 +40,7 @@ class Order < ActiveRecord::Base
     ])
   }
 
-  scope :latest, -> { order('id desc') }
+  scope :descending, -> { order('id desc') }
```
Comment (steveyken) [inline app/models/order.rb]: :+1: good clarification

File: spec/controllers/api/v1/orders_controller_spec.rb [hunk 5]
```diff
@@ -69,12 +83,55 @@
...
       it 'returns empty response if search text is draft order' do
-        request.headers["X-GOODCITY-APP-NAME"] = "stock.goodcity"
         get :index, searchText: draft_order.code
         expect(response.status).to eq(200)
         expect(parsed_body['designations'].count).to eq(0)
         expect(parsed_body['meta']['total_pages']).to eql(0)
       end
+
+      it 'can search orders using their description (case insensitive)' do
+        order = FactoryBot.create :order, :with_state_submitted, description: 'IPhone 100s'
```
Comment (steveyken) [inline spec/controllers/api/v1/orders_controller_spec.rb]: :smile:

Comment (namrataukirde) [reply spec/controllers/api/v1/orders_controller_spec.rb]: 😆

**Comment by namrataukirde** [comment]
Nice work @patrixr  with specs. Thanks for changes. :)

**Pull request merged**